tests/client/test_auth.py

```diff
@@ -179,34 +180,37 @@ def auth(request):
     }
 
 
-def test_auth_hidden_url():
+def test_auth_hidden_url() -> None:
     url = "http://example-username:example-password@example.org/"
     expected = "URL('http://example-username:[secure]@example.org/')"
     assert url == URL(url)
     assert expected == repr(URL(url))
 
 
 @pytest.mark.asyncio
-async def test_auth_hidden_header():
+async def test_auth_hidden_header() -> None:
     url = "https://example.org/"
     auth = ("example-username", "example-password")
 
     client = Client(dispatch=MockDispatch())
     response = await client.get(url, auth=auth)
 
-    assert "'authorization': '[secure]'" in str(response.request.headers)
+    request = typing.cast(Request, response.request)
```

Comment:
This might prove to be bad UX for our users: `response.request` is *almost always* set, *except* for some tests where *we* create a `Response` without a request. May look into those situations and then turn on `request` as non-optional on `Response`.

Reply:
Good call. Let's defer this until #666 is in, and we can drop the cast.